modules/javafx.graphics/src/main/native-glass/mac/GlassViewDelegate.m

```diff
@@ -1042,30 +1042,27 @@ - (void)startDrag:(NSDragOperation)operation withItems:(NSArray<NSDraggingItem*>
                     [image setSize:NSMakeSize(rect.size.width, rect.size.height)];
                 }
             }
-        }
-
-        if (image != nil)
-        {
-            // select the center of the image as the drag origin
-            // TODO http://javafx-jira.kenai.com/browse/RT-17629
-            // would be nice to get this info from the Java layer,
-            // so that we could adjust the drag image origin based on where in the src it was clicked on
-            dragPoint.x -= ([image size].width/2.0f);
-            dragPoint.y -= ([image size].height/2.0f);
 
             NSString *offsetString = [pbItem stringForType:DRAG_IMAGE_OFFSET];
             if (offsetString != nil) {
                 NSPoint offset = NSPointFromString(offsetString);
                 //Adjust offset to the image size
-                float imageHalfX = [image size].width/2.0f;
-                float imageHalfY = [image size].height/2.0f;
+                float imageW = [image size].width;
+                float imageH = [image size].height;
 
-                if (offset.x > imageHalfX || offset.x < -imageHalfX) {
-                    offset.x = imageHalfX * (offset.x > 0 ? 1 : -1);
+                if (offset.x < 0.0f) {
```

Comment:
That seems out of scope for this PR. We typically don't do any special processing to deal with NaN unless there is a problem. If needed, maybe it could be done as part of the follow-up bug [JDK-8338204](https://bugs.openjdk.org/browse/JDK-8338204).

Reply:
Sounds good, especially since it seems to treat NaNs as 0's.